src/util.rs

```diff
@@ -62,7 +62,14 @@ fn exit_with_code(r: reqwest::StatusCode) {
 // todo : assume https as the default scheme
 // Or get rid of this.
 pub fn url_validation(url: &str) -> Result<Url> {
-    Url::parse(url).context(format!("URL args: \'{}\' is not valid", url))
+    match Url::parse(url) {
+        Ok(pass) => {
+            Ok(pass)
+        },
+        Err(_error) => {
+            Url::parse(&format!("https://{}",url)).context(format!("URL args: \'{}\' is not valid", url))
+        },
+    }
 }
```

Comment:
One neat way to avoid the `Ok(pass)` duplication is to only map the error : 

```suggestion
    Url::parse(url).or_else(|_| {
        Url::parse(&format!("https://{}", url))
            .context(format!("URL arg: \'{}\' is not valid", url))
    })
}
```

Reply:
Thanks a lot. Surely learn to use Rust doc : )